.travis.yml

```diff
@@ -5,39 +5,41 @@ env:
     secure: "qFfvf0YAK5NWkH5fj708Z+1qIHtOFrmNhslB+n14rt5J3fJjY0A/4lbLbhx2jACYME+OU2m66maxh5abEZCF4ZGi4zpJa01IUYesTTa3qdM9kozB2TfFcl0d1xffuGpAh9kPchaadCvx/BU9fYFzkd8CLD5mQWCybwqM+6xpoO/KULDXhlcOPGbzg65xmPJANZH+ldH0NgfVLqofeFUg1j0bke1n+7DLf+Oh4ZiYZavciJjhuYx561TkkSEd1gqQGImEfkmKmhz82E8curx6uJW+kD9WagNbLj4FvEsBETIM7BSE8sei2vOVO42OnzvCej+P98IUelEqBS6sTeUNp79f8cC8n7nSYUJwTpshturBFoIFfcVQNoebdaao6gWpnxhUEnsc36AMl0DpIT0DsA9SxZQRjqhAEZmAWmASJMc5t/jWIjmJrqQxDy80HGJPEhIMTOAZfAiFqwyQwYTvJ0tOTW+jk2P9oriLHwi0Pr88uPGD5HWS6ULPoAutQAWMKCYgb5PoOiB8vbUZT9RTXfdLkHLkpw0ICaj5sam2o32CMLm8KZzfNVLJ/kUgdnkYUpC0vo+PGfguiJX8fvZxB06+niljW9iHteqVuoP1tADQwby4jS2ob3eL7TgCGuGk6gQkvHkLmOzlagKeXu1YZ0ow5vq6C+vSEfKEzIDrhXk="
 
+services:
+  - docker
+
 # todo: remove this once travis fixes their upstream issue (travis-ci/travis-ci#8836)
 sudo: required
-addons:
-  chrome: stable
 
-script:
-  # Bazel build & test everything. Can't use `bazel build ...` yet, because
-  # javaharness doesn't build on Travis (needs working Android SDK).
-  - >
-    bazel build --noshow_progress --noshow_loading_progress
-    //particles/...
-    //src/...
-    //src_kt/...
-  - >
-    bazel test --noshow_progress --noshow_loading_progress --test_output=errors
-    //particles/...
-    //src/...
-    //src_kt/...
-  - npm run test-with-start
-  - npm run build:rollup
-  - npm --prefix=server test
+# Bazel build & test everything. Can't use `bazel build ...` yet, because
+# javaharness doesn't build on Travis (needs working Android SDK).
+jobs:
+  include:
+    - stage: build
+      script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel build --disk_cache=/disk-cache/ //particles/... //src/... //src_kt/... --verbose_failures"
+    - stage: test
+      script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //particles/..."
+      name: "Bazel Test: //particles/..."
+    - script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest/bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //src/..."
+      name: "Bazel Test: //src/..."
+    - script: docker run -v $(pwd):/usr/src/app/ -v /home/travis/disk-cache:/disk-cache/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; bazel test --disk_cache=/disk-cache/ //src_kt/..."
+      name: "Bazel Test: //src_kt/..."
+    - script: docker run -v $(pwd):/usr/src/app/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; npm run test-no-web"
+      name: "Node Test: Unit"
+    - script: docker run -v $(pwd):/usr/src/app/ alxrsngrtn/arcs-travis-build:latest /bin/sh -c "cd /usr/src/app; npm run test-shells"
+      name: "Node Test: Shells"
 
 cache:
   npm: true
   directories:
-    - /home/travis/.cache/bazel/
+    - /home/travis/disk-cache/
```

Comment:
Does this cache hold the docker config?

Reply:
No, it shouldn't.

Reply:
In that case, I'm not sure how the bazel cache would reach disk-cache. Please make it clear if it should, how and if not, why.

Thank you

Reply:
Added explainer comments.